langchain/vectorstores/base.py

```diff
@@ -107,6 +107,12 @@ def similarity_search(
     ) -> List[Document]:
         """Return docs most similar to query."""
 
+    @abstractmethod
+    def similarity_search_by_text(
```

Comment:
could we remove this off base class? don't think this generalizes to all vector stores